src/coap_session.c

```diff
@@ -190,7 +190,10 @@ void coap_session_free(coap_session_t *session) {
 
 size_t coap_session_max_pdu_size(const coap_session_t *session) {
   size_t max_with_header = (size_t)(session->mtu - session->tls_overhead);
-  if (!COAP_DISABLE_TCP || COAP_PROTO_NOT_RELIABLE(session->proto))
+#if COAP_DISABLE_TCP
```

Comment:
This is actually a bugfix. Maybe this should at least be highlighted in the commit message? Currently it just speaks of dead code removal.

Reply:
I have just re-worked this change to remove the dead TCP workings.